scripts/profiler/real_time_plot.py

```diff
@@ -54,46 +53,51 @@ def main():
+    # Events are made to ensure that PlotNordic and ModelCreator classes are initialized before Rtt2Stream starts sending data.
     event_plot = Event()
     event_model_creator = Event()
+    # Setting these events results in closing corresponding modules.
+    event_close_rtt2stream = Event()
+    event_close_model_creator = Event()
+    event_close_plot = Event()
 
     streams = Stream.create_stream(3)
 
     try:
         processes = []
-        processes.append(Process(target=rtt2stream,
-                                 args=(streams[0], event_plot, event_model_creator,
-                                       log_lvl_number),
-                                 daemon=True))
-        processes.append(Process(target=model_creator,
-                                 args=(streams[1], event_model_creator,
-                                       args.dataset_name, log_lvl_number),
-                                 daemon=True))
-        processes.append(Process(target=dynamic_plot,
-                                 args=(streams[2], event_plot, log_lvl_number),
-                                 daemon=True))
+        processes.append((Process(target=rtt2stream,
+                                  args=(streams[0], event_plot, event_model_creator,
+                                        event_close_rtt2stream, log_lvl_number),
+                                  daemon=True),
+                          event_close_rtt2stream))
+        processes.append((Process(target=model_creator,
+                                  args=(streams[1], event_model_creator, event_close_model_creator,
+                                        args.dataset_name, log_lvl_number),
+                                  daemon=True),
+                          event_close_model_creator))
+        processes.append((Process(target=dynamic_plot,
+                                  args=(streams[2], event_plot, event_close_plot, log_lvl_number),
+                                  daemon=True),
+                          event_close_plot))
 
-        for p in processes:
+        for p, _ in processes:
             p.start()
-
         is_waiting = True
         while is_waiting:
-            for p in processes:
+            for p, _ in processes:
                 p.join(timeout=0.5)
                 # Terminate other processes if one of the processes is not active.
                 if len(processes) > len(active_children()):
                     is_waiting = False
                     break
-
-        for p in processes:
+        for p, event_close in processes:
             if p.is_alive():
-                os.kill(p.pid, signal.SIGINT)
+                event_close.set()
                 # Ensure that we stop processes in order to prevent profiler data drop.
                 p.join()
 
     except KeyboardInterrupt:
-        for p in processes:
+        for p, _ in processes:
```

Comment:
I would consider setting the close event to ensure that every process would gracefully shutdown without relying on `KeyboardInterrupt`. We could even consider ignoring the `KeyboardInterrupt` in the child processes in the future to ensure that order of shutdown is proper.

Reply:
For me it would be strange to set close event here even though we know that it won't be read in submodules processes. Anyway, ignoring `KeyboardInterrupt` in the child processes seems to to be good idea.

Reply:
Well, actually it appears that it's impossible to ignore exception in python in a way that you can put a block of code inside some `try:` and expect it not to interrupt its work when exception occur. I don't know whether it's what you meant, that's how I understand your suggestion. In that case, I would leave it as it is now since there is not much we can do. We could make some events that process could share and look at them when receiving `KeyboardInterrupt`, but for me it can possibly only make the code more complicated. Ending script with keyboard interrupt is not a dedicated way to close the script, so what I think we have to achieve is to make sure that collected data is not lost and that's all. Also there are some problems with closing scripts with keyboard interrupt on windows, because plot module spawns some child processes and they also receive interrupts. @MarekPieta what do you think about it?

Reply:
It seems like we could use `signal` library to capture SIGINT in main module (like `data_collector`, `real_time_plot`). In that case the SIGINT signal is not directly propagated to child processes and we should be able to close them in proper order setting events. That would prevent data loss and it would assure that we are always performing the close operation in the same way.

If you find some time tomorrow morning, we could discuss the solution and perform a few checks together (I already verified that main process could catch SIGINT and no longer forward it to the child processes).